RedHatInsights/frontend-components
# feat(RHIF-258): Support resource definitions in RBACHook

Implements https://issues.redhat.com/browse/RHIF-258.

This makes it possible to verify resource definitions while evaluating RBAC permissions. The backward compatibility is ensured by parameter `checkResourceDefinitions` which is set to false by default. This way, apps can opt in for the more granular checks only when they have this new parameter defined.

- [x] Update doesHavePermissions function
- [x] Update hasAllPermissions function
- [x] Ensure functions are backward compatible (won't make apps fail)

## How to test

To make sure the new version doesn't break apps, you can run this both modified packages (e.g., with `npm link @redhat-cloud-services/frontend-components @redhat-cloud-services/frontend-components-utilities`) together with any app locally and check that all pages work and there are no authorization issues.

To test resource definitions, it's somewhat trickier since only a few apps support it (like cost management). You again can run this package together with the selected app and try to play and see whether resource definitions are respected.
